benchmark/sqllogictest_tester/run_tester.sh

```diff
@@ -0,0 +1,273 @@
+if [ -z "$DOLT_CREDS" ]; then fail Must supply DOLT_CREDS; fi
+if [ -z "$CREDS_HASH" ]; then fail Must supply CREDS_HASH; fi
+if [ -z "$TMP_TESTING_DIR" ]; then fail Must supply TMP_TESTING_DIR; fi
+if [ -z "$TMP_CSV_DIR" ]; then fail Must supply TMP_CSV_DIR; fi
+if [ -z "$TEST_FILE_DIR_LIST" ]; then fail Must supply TEST_FILE_DIR_LIST; fi
+if [ -z "$COMMITS_TO_TEST" ]; then fail Must supply COMMITS_TO_TEST; fi
+if [ -z "$DOLT_BRANCH" ]; then fail Must supply DOLT_BRANCH; fi
+if [ -z "$TEST_N_TIMES" ]; then fail Must supply TEST_N_TIMES; fi
+
+[[ "$TEST_N_TIMES" =~ ^[0-9]+$ ]] || fail TEST_N_TIMES must be a number
+
+function setup() {
+    rm -rf "$CREDSDIR"
+    mkdir -p "$CREDSDIR"
+    dolt config --global --add metrics.disabled true
+    dolt creds import "$DOLT_CREDS"
+    dolt version
+}
+
+function setup_testing_dir() {
+    rm -rf "$TMP_TESTING_DIR"
+    mkdir -p "$TMP_TESTING_DIR"
+
+    IFS=', ' read -r -a test_list <<< "$TEST_FILE_DIR_LIST"
+    for fd in "${test_list[@]}"
+    do
+         dir=$(dirname "$fd")
+         mkdir -p "$TMP_TESTING_DIR"/"$dir"
+         cp -r "$sqllogictest_checkout"/test/"$fd" "$TMP_TESTING_DIR"/"$fd"
+    done
+}
+
+function checkout_branch_if_exists() {
+  local cmd="$1"
+  local branch_name="$2"
+  local hash="$3"
+
+  if [ `"$cmd" branch --list "$branch_name" | wc -l` -eq 1 ]; then
+    "$cmd" checkout "$branch_name"
+  else
+    if [ -z "$hash" ]; then
+      "$cmd" checkout -b "$branch_name";
+    else
+      "$cmd" checkout -b "$branch_name" "$hash";
+    fi
+  fi
+}
+
+function with_dolt_commit() {
+  local commit_hash="$1"
+    (
+      cd "$base_dir"/tempDolt/go
+      git checkout "$commit_hash"
+
+      checkout_branch_if_exists "git" "temp-$commit_hash" "$commit_hash"
+
+      git log -n 1
+
+      if ! [ -x "$base_dir"/.ci_bin/"$commit_hash"/dolt ]; then
+          if ! [ -d "$base_dir"/.ci_bin ]; then
+            mkdir -p "$base_dir"/.ci_bin/"$commit_hash"
+          fi
+
+          go build -mod=readonly -o "$base_dir"/.ci_bin/"$commit_hash"/dolt ./cmd/dolt/.
+      fi
+    )
+
+    echo "Finished installing dolt from $commit_hash:"
+    export PATH="$base_dir/.ci_bin/$commit_hash":$old_path
+    dolt version
+}
+
+function with_dolt_checkout() {
+    (
+      cd "$base_dir"/go
+      if ! [ -x "$base_dir"/.ci_bin/checkout/dolt ]; then
+          if ! [ -d "$base_dir"/.ci_bin ]; then
+            mkdir -p "$base_dir"/.ci_bin/checkout
+          fi
+          go get -mod=readonly ./...
+          go build -mod=readonly -o "$base_dir"/.ci_bin/checkout/dolt ./cmd/dolt/.
+      fi
+    )
+    echo "Finished installing dolt from checkout:"
+    export PATH="$base_dir/.ci_bin/checkout":$old_path
+    dolt version
+}
+
+function import_once() {
+    local commit_hash="$1"
+    local test_num="$2"
+    local parsed="$log_dir/parsed-$commit_hash-$test_num".json
+
+    dolt checkout "$DOLT_BRANCH"
+
+    checkout_branch_if_exists "dolt" "temp-$commit_hash" ""
```

Comment:
for the `dolt` case, im using the shallow branch (DOLT_BRANCH) as a root to fork from. on the children branches from this root, i import the test results data for each committer, and Im just using fresh child `dolt` branches for each import, instead of importing the data from each commiter into the `nightly` table of branch DOLT_BRANCH... 🤠 I hope that explanation makes sense lol

Reply:
also found a `dolt` panic! `dolt checkout ""` and `dolt checkout -b ""` both cause `index out of range` panics... will make `bats` test.